### ISSUE TITLE
Click Event on bar (bar chart) or pie (pie chart)

### ISSUE THREAD

**Issue by bishinew** [PhilJay/MPAndroidChart#609]
Hi Phil

We have an app with Pie and Bar chart. When user clicks a bar or a pie, it will fire a listener/event based on which bar or pie is clicked. How to do this? Can you share some sample code?

Many thanks

**Comment by nbau21** [comment]
Use these:

[Pie chart listener](https://github.com/PhilJay/MPAndroidChart/blob/f629df992b0b1200bef8c873cbc3b4d4c827f61e/MPChartLib/src/com/github/mikephil/charting/listener/PieRadarChartTouchListener.java)

[Bar chart listener](https://github.com/PhilJay/MPAndroidChart/blob/f629df992b0b1200bef8c873cbc3b4d4c827f61e/MPChartLib/src/com/github/mikephil/charting/listener/BarLineChartTouchListener.java)

Basically:

        mChart.setOnChartValueSelectedListener( new OnChartValueSelectedListener() {
            @Override
            public void onValueSelected(Entry e, int dataSetIndex, Highlight h) {
                //fire up event
            }

            @Override
            public void onNothingSelected() {

            }
        });

**Comment by PhilJay** [comment]
Take a look at the wiki: https://github.com/PhilJay/MPAndroidChart/wiki/Interaction-with-the-Chart

**Comment by bishinew** [comment]
Hi
Thanks for your suggestion. We plan to use the OnChartValueSelectedListener in the ViewPager Fragment. We encounter a problem: The pager swipe gesture always trigger the OnChartValueSelectedListener. Is there anything we can do to click the bar or bie to trigger OnChartValueSelectedListener but swipe gesture to switch pages?
Many thanks.
Eric 
Date: Fri, 1 May 2015 01:16:31 -0700

**Comment by bishinew** [comment]
Hi Developer,
We reported the issue on swiping that triggers the select value listener. We did more tests on this. Version 2.0.7 seems working OK. But version 2.1.0 has this problem. Can you please take a look at it. If you need more information. Please let us know.
ThanksEric
From: zhoueric@hotmail.com
To: reply@reply.github.com; mpandroidchart@noreply.github.com
Subject: RE: [MPAndroidChart] Click Event on bar (bar chart) or pie (pie chart) (#609)
Date: Mon, 11 May 2015 16:36:26 -0500




Hi
Thanks for your suggestion. We plan to use the OnChartValueSelectedListener in the ViewPager Fragment. We encounter a problem: The pager swipe gesture always trigger the OnChartValueSelectedListener. Is there anything we can do to click the bar or bie to trigger OnChartValueSelectedListener but swipe gesture to switch pages?
Many thanks.
Eric 
Date: Fri, 1 May 2015 01:16:31 -0700

**Comment by sreenath2406** [comment]
I'm facing the same issue.
Create a viewpager and add BarChart in one of the fragment. On Swipe, fragment is not changed because, OnChartValueSelectedListener is triggered over swipe of ViewPager.
Can we restrict the touch for bar graph to the height of bar ?

**Comment by Sannaraek** [comment]
hoping that there is a OnLongClick listener for graph data, would be a sure fix. Although, anybody here know how to initiate it?

**Comment by ValeriyBugaev** [comment]
Good day. How intercept center click on PieChart? (where total values displayed)
Thanks.

**Comment by ravirajjak** [comment]
does any one know how to show detail values if I select particular point.
For eg: in line chart if I click on particular point it should show details regarding that point with date time and all other values
